Makefile

```
@@ -32,7 +32,24 @@ gxdr/xdr_generated.go: $(XDRS)
 	go run github.com/xdrpp/goxdr/cmd/goxdr -p gxdr -enum-comments -o $@ $(XDRS)
 	go fmt $@
 
-xdr/xdr_generated.go: $(XDRS) Rakefile Gemfile.lock
-	bundle exec rake xdr:generate
+define xdrheader
+//lint:file-ignore S1005 The issue should be fixed in xdrgen. Unfortunately, there's no way to ignore a single file in staticcheck.
+//lint:file-ignore U1000 fmtTest is not needed anywhere, should be removed in xdrgen.
+endef
+export xdrheader
+
+xdr/xdr_generated.go: $(XDRS)
+	docker run --platform linux/amd64 -it --rm -v $$PWD:/wd -w /wd ruby /bin/bash -c '\
+		gem install specific_install && \
```

Could you lock this gem install to a specific version to ensure consistency.
```suggestion
		gem install specific_install -v 0.3.7 && \
```

Fun fact, this still doesn't guarantee a consistent install of dependencies, and there's no way with `gem` to make it consistent. (See https://twitter.com/___leigh___/status/1520997579772678144.) It doesn't matter for specific_install though because it has no dependencies.